vpr/src/timing/VprTimingGraphResolver.cpp

```diff
@@ -197,3 +224,109 @@ e_timing_report_detail VprTimingGraphResolver::detail_level() const {
+            net_component.type_name = device_ctx.rr_nodes[node->inode].type_string(); //write the component's type as a routing resource node
+            net_component.type_name += ":" + std::to_string(node->inode) + " ";       //add the index of the routing resource node
+            if (device_ctx.rr_nodes[node->inode].type() == OPIN || device_ctx.rr_nodes[node->inode].type() == IPIN) {
+                net_component.type_name += "side:";                                        //add the side of the routing resource node
+                net_component.type_name += device_ctx.rr_nodes[node->inode].side_string(); //add the side of the routing resource node
+                // For OPINs and IPINs the starting and ending coordinate are identical, so we can just arbitrarily assign the start to larger values
+                // and the end to the lower coordinate
+                start_x = std::to_string(device_ctx.rr_nodes[node->inode].xhigh()); //start coordinates have larger value
+                start_y = std::to_string(device_ctx.rr_nodes[node->inode].yhigh());
+                end_x = std::to_string(device_ctx.rr_nodes[node->inode].xlow()); //end coordinates have smaller value
+                end_y = std::to_string(device_ctx.rr_nodes[node->inode].ylow());
+            }
+            if (device_ctx.rr_nodes[node->inode].type() == CHANX || device_ctx.rr_nodes[node->inode].type() == CHANY) {                                               //for channels, we would like to describe the component with segment specific information
+                net_component.type_name += device_ctx.arch->Segments[device_ctx.rr_indexed_data[device_ctx.rr_nodes[node->inode].cost_index()].seg_index].name + " "; //Write the segment name
+                net_component.type_name += "length:" + std::to_string(device_ctx.rr_nodes[node->inode].length());                                                     //add the length of the segment
+                //Figure out the starting and ending coordinate of the segment depending on the direction
+
+                if (device_ctx.rr_nodes[node->inode].direction() == DEC_DIRECTION) {    //signal travels along decreasing direction
+                    start_x = std::to_string(device_ctx.rr_nodes[node->inode].xhigh()); //start coordinates have larger value
+                    start_y = std::to_string(device_ctx.rr_nodes[node->inode].yhigh());
+                    end_x = std::to_string(device_ctx.rr_nodes[node->inode].xlow()); //end coordinates have smaller value
+                    end_y = std::to_string(device_ctx.rr_nodes[node->inode].ylow());
+                }
+
+                else if (device_ctx.rr_nodes[node->inode].direction() == INC_DIRECTION || device_ctx.rr_nodes[node->inode].direction() == NO_DIRECTION || device_ctx.rr_nodes[node->inode].direction() == BI_DIRECTION) {
```

Comment:
Since only the DEC_DIRECTION case is different, consider making this just an `else` condition.